src/Bicep.Core/Semantics/Namespaces/SystemNamespaceSymbol.cs

```diff
@@ -389,6 +391,67 @@ public class SystemNamespaceSymbol : NamespaceSymbol
                 .WithReturnType(LanguageConstants.String)
                 .WithDescription("Returns a value in the format of a globally unique identifier. **This function can only be used in the default value for a parameter**.")
                 .WithFlags(FunctionFlags.ParamDefaultsOnly)
+                .Build(),
+
+            new FunctionOverloadBuilder("loadContent")
+                .WithReturnType(LanguageConstants.String)
+                .WithDescription("Load content of specified file into a string. Content loading occurs during compilation, not in runtime. Maximum allowed content size is 24,576 characters (including line endings).")
+                .WithRequiredParameter("filePath", LanguageConstants.String, "File path which content will be loaded")
+                .WithOptionalParameter("encoding", LanguageConstants.String, "File encoding. If not provided, Bicep will try to automatically detect the encoding of a file based on the presence of byte order marks.")
+                .WithDiagnosticsCollector((binder, diagnostics, arguments) => {
+                    if (arguments[0].syntax.Expression is StringSyntax filePathSyntax)
+                    {
+                        if (filePathSyntax.IsInterpolated())
+                        {
+                            diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0].syntax).CompileTimeConstantRequired());
+                            return;
+                        }
+
+                        var filePath = filePathSyntax.TryGetLiteralValue() ?? string.Empty;
```

Comment:
>TryGetLiteralValue [](start = 54, length = 18)

Is it actually possible for `TryGetLiteralValue()` to return `null` here? If not, why not just do `?? throw ...`

Reply:
not really, perhaps `!` would work.

Reply:
We try to avoid it on the product side. (Fair game in tests, though.)

---
In reply to: [599025682](https://github.com/miqm/bicep/pull/1#discussion_r599025682) [](ancestors = 599025682)